### PR TITLE
feat: All unary operators

### PR DESCRIPTION
**Pull request by vasantha-helprack**
# Added
- `.gitignore` changes for GNU Global and Emacs
- New operators:
  - Exp
  - Sqrt
  - Sin
  - Log
  - Tanh
  - Neg
  - Sigmoid
  - Abs
 
# Changed
- Refactored operators:
  - Relu
# Test:
- [x] Tested with internal models.
- [x] Tested with Anushree's models.
- [x] All operators test passed

### CODE REVIEW

File: src/Conversion/ONNXToTorch/NN/ElementwiseOp.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    Location loc = unaryOp.getLoc();
+    Value operand = unaryOp.input();
+    mlir::MLIRContext *context =  unaryOp.getContext();
+
+    auto operandTensorType = operand.getType().template dyn_cast<TensorType>();
+    TensorType resultTensorType = unaryOp.getResult().getType().template dyn_cast<TensorType>();
+
+    auto operandType = Torch::ValueTensorType::get(context,
+                                           operandTensorType.getShape(),
+                                           operandTensorType.getElementType());
```
Comment (oowekyala) [inline src/Conversion/ONNXToTorch/NN/ElementwiseOp.cpp:43]: It seems like we do this quite often, here and in conversions for other ops. Would it make sense to factorize this into a free standing function somewhere?
```c++
Torch::ValueTensorType toTorchType(mlir::MLIRContext ctx, Type t) {
   auto type = t.template dyn_cast<TensorType>();
   return Torch::ValueTensorType(ctx, type.getShape(), type.getElementType());
}
// then use like
auto operandType = toTorchType(context, unaryOp.getOperand().getType());
auto resultType = toTorchType(context, unaryOp.getResult().getType());

**Review by oowekyala — COMMENTED** [review]
I think this is a nice generic solution and am happy it can be used for many ops without changes. I'd like my comment about the duplicate implementation and tests to be addressed before merging though

File: src/Conversion/ONNXToTorch/NN/ElementwiseOp.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                 << "\n";
+
+    rewriter.replaceOpWithNewOp<TensorStaticInfoCastOp>(op, resultType, result);
+
+    return success();
+  }
+};
+
+template <typename ONNXUnaryOp, typename TorchUnaryOp>
+struct ONNXToTorchElementwiseUnaryOpLowering2 : public ConversionPattern {
```
[the remark this answers is not in the document]
Comment (vasantha-helprack) [reply src/Conversion/ONNXToTorch/NN/ElementwiseOp.cpp]: Thanks for pointing out `getOperand`. I've fixed code duplication. I'll try to add tests as well.

File: src/Conversion/ONNXToTorch/NN/ElementwiseOp.cpp (hunk 1 above)
Comment (vasantha-helprack) [reply src/Conversion/ONNXToTorch/NN/ElementwiseOp.cpp]: Done

Comment (oowekyala) [reply src/Conversion/ONNXToTorch/NN/ElementwiseOp.cpp]: Thanks

**Review by oowekyala — APPROVED** [review]
LGTM, thanks

**Review by BraunPhilipp — APPROVED** [review]

**Pull request merged**